### PR TITLE
feat: add simulator driver support

### PR DESCRIPTION
**Pull request by JMolenkamp**

### CODE REVIEW

File: src/Atc.Kepware.Configuration/KepwareContracts/Connectivity/Simulator/SimulatorDevice.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+namespace Atc.Kepware.Configuration.KepwareContracts.Connectivity.Simulator;
+
+internal class SimulatorDevice : DeviceBase, ISimulatorDevice
```
Comment (perkops) [inline src/Atc.Kepware.Configuration/KepwareContracts/Connectivity/Simulator/SimulatorDevice.cs:3]: Are we missing SimulatorDeviceIdFormat here?

**Comment by perkops** [comment]
LGTM - a few minor things and questions added :)

File: src/Atc.Kepware.Configuration.Contracts/Connectivity/Drivers/Simulator/SimulatorDevice.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+namespace Atc.Kepware.Configuration.Contracts.Connectivity.Drivers.Simulator;
+
+public sealed class SimulatorDevice : DeviceBase, ISimulatorDevice
```
[the remark this answers is not in the document]
Comment (JMolenkamp) [reply src/Atc.Kepware.Configuration.Contracts/Connectivity/Drivers/Simulator/SimulatorDevice.cs:3]: Keenly noted!

Definitely the easier solution, and fine by me. 
However, wouldn't it be cleaner to introduce an `ISimulatableDeviceBase` interface that derives from `IDeviceBase` and defines the `Simulated` property?

[the remark this answers is not in the document]
Comment (JMolenkamp) [reply src/Atc.Kepware.Configuration.Contracts/Connectivity/Drivers/Simulator/SimulatorDevice.cs:3]: Yeah, it should.

Initially, I wanted to omit it completely. It only seems to affect the way the id is displayed in the kepserver configuration UI. For me, no use case will ever exist where I would want to get/set this property. Someone else might though, so I added it, but apparently somewhat half-heartedly.

**Comment by JMolenkamp** [comment]
Question: Should I rewrite PR history by force pushing like I did here, or is it preferable to add additional commits?

**Comment by perkops** [comment]
> Question: Should I rewrite PR history by force pushing like I did here, or is it preferable to add additional commits?

Im fine with both - whatever makes sense. If its just fixing something, that still makes sense without a "train-of-though", then force-push is fine, otherwise additional commits :)

File: src/Atc.Kepware.Configuration.Contracts/Connectivity/Drivers/Simulator/SimulatorDevice.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (perkops) [reply src/Atc.Kepware.Configuration.Contracts/Connectivity/Drivers/Simulator/SimulatorDevice.cs:3]: That could work aswell

[the remark this answers is not in the document]
Comment (JMolenkamp) [reply src/Atc.Kepware.Configuration.Contracts/Connectivity/Drivers/Simulator/SimulatorDevice.cs:3]: Keeping it as suggested for now, just easier 😇

**Comment by JMolenkamp** [comment]
I'm not really sure if resolving the conversations was up to me

**Review by perkops — APPROVED** [review]

**Pull request merged**